src/test/ui/consts/ptr_comparisons.rs

```diff
@@ -0,0 +1,46 @@
+// compile-flags: --crate-type=lib
+
+#![feature(const_panic, core_intrinsics, const_raw_ptr_comparison, const_ptr_offset)]
+
+const FOO: &usize = &42;
+
+macro_rules! check {
+    (eq, $a:expr, $b:expr) => {
+        pub const _: () =
+            assert!(std::intrinsics::ptr_guaranteed_eq($a as *const u8, $b as *const u8));
+    };
+    (ne, $a:expr, $b:expr) => {
+        pub const _: () =
+            assert!(std::intrinsics::ptr_guaranteed_ne($a as *const u8, $b as *const u8));
+    };
+    (!eq, $a:expr, $b:expr) => {
+        pub const _: () =
+            assert!(!std::intrinsics::ptr_guaranteed_eq($a as *const u8, $b as *const u8));
+    };
+    (!ne, $a:expr, $b:expr) => {
+        pub const _: () =
+            assert!(!std::intrinsics::ptr_guaranteed_ne($a as *const u8, $b as *const u8));
+    };
+}
+
+check!(eq, 0, 0);
+check!(ne, 0, 1);
+check!(!eq, 0, 1);
+check!(!ne, 0, 0);
+check!(ne, FOO as *const _, 0);
+check!(!eq, FOO as *const _, 0);
+check!(!eq, FOO as *const _, FOO as *const _);
+check!(ne, unsafe { (FOO as *const usize).offset(1) }, 0);
+check!(!eq, unsafe { (FOO as *const usize).offset(1) }, 0);
+const _: *const usize = unsafe { (FOO as *const usize).offset(2) };
+//~^ NOTE
```

Comment:
Please add a comment explaining what this tests. It doesn't test the comparison function.

Reply:
Your new comments still do not explain what this tests and why it exists in this file.

Reply:
I added a new comment explaining both the reason and the actions that should be taken

Reply:
This is great, thanks. :)